lib/rules/require-dot-notation.js

```diff
@@ -62,30 +99,50 @@
  * var a = b.c;
  * var a = b[c.d];
  * var a = b[1];
- * var a = b.while;
+ * var a = b['while']; // reserved word in ES3
  * ```
  *
  * ##### Invalid
  *
  * ```js
  * var a = b['c'];
- * var a = b['while']; // reserved words can be property names in ES5
  * ```
  */
 
 var assert = require('assert');
 var utils = require('../utils');
+var reservedWords = require('../reserved-words');
 
 module.exports = function() {};
 
 module.exports.prototype = {
 
     configure: function(options) {
+        if (typeof options !== 'object') {
+            assert(
+                options === true || options === 'except_snake_case',
+                this.getOptionName() + ' option requires either a true value or an object'
+            );
+
+            var _options = {};
+            if (options === 'except_snake_case') {
+                console.warn('Value "except_snake_case" was deprecated\n' +
```

Comment:
I think we should remove warn completely, but when we start warn about such things, we should do it in a general way, so we wouldn't need to add deprecation noticed in every rule.

Just have file with the old way and the new way. @mdevils already working on the mapping file.

Besides, never use `console` in rules, since such errors are not consumable through reporters.